pkg/policy/execute.go

```diff
@@ -415,36 +415,51 @@ func (e *Executor) deleteViews(ctx context.Context, policy *Policy) {
 	}
 }
 
-func GenerateExecutionResultFile(result *ExecutionResult, outputDir string) error {
+func GenerateExecutionResultFile(result *ExecutionResult, outputDir string) diag.Diagnostics {
 	fs := afero.NewOsFs()
 
 	if err := fs.MkdirAll(outputDir, 0755); err != nil {
-		return err
+		return diag.FromError(
+			err,
+			diag.USER,
+			diag.WithDetails(fmt.Sprintf("failed to create directory \"%s\"", outputDir)),
+		)
 	}
 
 	// result.PolicyName is the full selector for this policy run.
 	// The name of the output file should just be the base policy.
 	// e.g. for "aws//cis_v1.2.0", the output file should be "aws.json"
 	basePolicyName, err := extractFirstPathComponent(result.PolicyName)
 	if err != nil {
-		return err
+		return diag.FromError(err, diag.INTERNAL)
 	}
 
-	f, err := fs.Create(fmt.Sprintf("%s.json", filepath.Join(outputDir, basePolicyName)))
+	filePath := filepath.Join(outputDir, basePolicyName)
+
+	f, err := fs.Create(fmt.Sprintf("%s.json", filePath))
 	if err != nil {
-		return err
+		return diag.FromError(
+			err,
+			diag.USER,
+			diag.WithDetails(fmt.Sprintf("failed to create file \"%s\"", filePath)),
```

Comment:
nit: Use `%q` instead of `\"%s\"`
Also wouldn't the `err` itself contain the file and the "failed to create" msg already?

Reply:
1) Done.
2) Not sure what are all the possible error messages from the library so opted for providing more context just in case. No big feelings here.